File: 0696. 计数二进制子串/遍历.cpp
// 36ms
class Solution {
public:
    int countBinarySubstrings(string s) {
        int len = s.size(), result = 0;
        int pre = 0, count = 0;
        char current = '0';
        for (auto c: s) {
            if (c == current) {
                count += 1;
            } else {
                result += min(pre, count);
                pre = count;
                count = 1;
                current = c;
            }
        }
        result += min(pre, count);
        return result;
    }
};

File: 0463. 岛屿的周长/遍历.swift
// 592ms
class Solution {
    func islandPerimeter(_ grid: [[Int]]) -> Int {
        var result = 0
        var pos = [(-1, 0), (1, 0), (0, -1), (0, 1)]
        var row = grid.count, col = row == 0 ? 0 : grid[0].count
        for i in 0..<row {
            for j in 0..<col {
                if grid[i][j] == 1 {
                    for (x, y) in pos {
                        let newX = i + x, newY = j + y
                        if (newX < 0 || newX >= row || newY < 0 || newY >= col) || grid[newX][newY] == 0 {
                            result += 1
                        }
                    }
                }
            }
        }
        return result
    }
}